cli/interactive_tests: more flakiness in test_missing_log_output.tcl

The following tests appear to have failed:

[#188699](https://teamcity.cockroachdb.com/viewLog.html?buildId=188699):

```
--- FAIL: acceptance/TestDockerCLI (114.870s)
null
--- FAIL: acceptance/TestDockerCLI/test_missing_log_output.tcl (30.020s)
cli_test.go:73: non-zero exit code: 1
--- FAIL: acceptance/TestDockerReadWriteBidirectionalReferenceVersion (17.120s)
reference_test.go:41: non-zero exit code: 1
--- FAIL: acceptance (proposer evaluated kv)/TestDockerReadWriteBidirectionalReferenceVersion (16.230s)
reference_test.go:41: non-zero exit code: 1

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/cli/interactive_tests/common.tcl
+++ b/pkg/cli/interactive_tests/common.tcl
@@ -22,7 +22,10 @@ set stty_init "cols 80 rows 25"
 proc eexpect {text} {
     expect {
 	$text {}
-	timeout {exit 1}
+	timeout {
+	    system "echo SERVER STDOUT; cat stdout.log; echo SERVER STDERR; cat stderr.log"
+	    exit 1
+	}
     }
 }
 
@@ -36,7 +39,7 @@ proc interrupt {} {
 # Preserves the invariant that the server's PID is saved
 # in `server_pid`.
 proc start_server {argv} {
-    system "mkfifo pid_fifo || true; $argv start --pid-file=pid_fifo --background & cat pid_fifo > server_pid"
+    system "mkfifo pid_fifo || true; $argv start --pid-file=pid_fifo --background >>stdout.log 2>>stderr.log & cat pid_fifo > server_pid"
 }
 proc stop_server {argv} {
     system "set -e; if kill -CONT `cat server_pid`; then $argv quit || true & sleep 1; kill -9 `cat server_pid` || true; else $argv quit || true; fi"

--- a/pkg/cli/interactive_tests/test_missing_log_output.tcl
+++ b/pkg/cli/interactive_tests/test_missing_log_output.tcl
@@ -60,10 +60,11 @@ start_server $argv
 
 # Now test `quit` as non-start command, and test that `quit` does not
 # emit logging output between the point the command starts until it
-# prints the final ok message.
-send "echo marker; $argv quit\r"
+# either prints the final ok message or fails with some error
+# (e.g. due to no definite answer from the server).
+send "echo marker; $argv quit 2>&1 | grep '^\\(\[IWEF\]\[0-9\]\\)' \r"
 set timeout 20
-eexpect "marker\r\nok\r\n:/# "
+eexpect "marker\r\n:/# "
 set timeout 5
 
 # Test quit as non-start command, this time with --logtostderr. Test

